crates/bevy_core/src/time/fixed_timestep.rs

```diff
@@ -1,301 +1,436 @@
-use crate::Time;
 use bevy_ecs::{
-    archetype::{Archetype, ArchetypeComponentId},
-    component::ComponentId,
-    query::Access,
     schedule::ShouldRun,
-    system::{IntoSystem, Res, ResMut, System},
-    world::World,
+    system::ResMut,
+    world::{FromWorld, World},
 };
-use bevy_utils::HashMap;
-use std::borrow::Cow;
+use bevy_utils::{Duration, Instant};
 
-/// The internal state of each [`FixedTimestep`].
-#[derive(Debug)]
-pub struct FixedTimestepState {
-    step: f64,
-    accumulator: f64,
+use crate::Time;
+
+/// The default step size used by [`FixedTimestep`].
+// 60Hz is a popular tick rate, but it can't be expressed as an exact float.
+// The nearby power of two, 64Hz, is more stable for numerical integration.
+pub const DEFAULT_TIMESTEP: Duration = Duration::from_micros(15625); // 64Hz
+
+/// Tracks how much time has advanced since its previous update and since the app was started.
+///
+/// It's just like [`Time`] (and internally coupled to it), except it advances in fixed increments.
```

Comment:
We should explicitly document the logic around what happens at various elapsed time intervals here. E.g:

- [0,1) time steps have elapsed:
- [1, 2) time steps have elapsed:
- [2, inf) time steps have elapsed:

Reply:
wdym?

`FixedTime` is there for systems running under `CoreSet::FixedUpdate` (I'll make note of this). It is advanced in "whole" increments (of the fixed timestep value), once every time we run `CoreSet::FixedUpdate` (just before it). 

Any "fractional" time accumulates on `FixedTimestepState` until there's enough to advance by a whole number.

Reply:
What happens if more than one timestep has elapsed? Do we lose a step?

Reply:
No, it's 1:1. `Time` advances every frame by `Time::delta` and we add the same amount to `FixedTimestepState`. `FixedTimestepState` then tries to "buy" steps with the time it's saved up. Each step costs "FixedTime::delta`.

If `FixedTimestepState` has steps purchased, we run all of them in a loop on the same frame.

Reply:
```suggestion
/// This resource can be used like [`Time`], but advances in discrete steps.
/// When [`Time`] advances, [`FixedTime`] also advances by the same amount.
/// If enough time has accumulated to buy a new step (controlled by [`FixedTime::delta`], it advances once,
/// accumulating any excess.
/// If a large amount of time has elapsed, this can cause multiple [`FixedTime`] steps to occur in a single pass of the [`Schedule`].
///
/// This resource is used to control whether or not systems that are part of  [`CoreSet::FixedUpdate`] run.
```